### DISCUSSION TITLE
GitHub App with read access to repository metadata can't list forks via REST API

### DISCUSSION THREAD

**Discussion by yury-imbro-ox** [community/community#84198] [Apps, API and Webhooks]
### Select Topic Area

Bug

### Body

Hi 👋

According to the documentation ([list-forks API](https://docs.github.com/en/rest/repos/forks?apiVersion=2022-11-28#list-forks), [repository permissions](https://docs.github.com/en/rest/authentication/permissions-required-for-github-apps?apiVersion=2022-11-28#repository-permissions-for-metadata)), GitHub Apps with read access to repository metadata should be able to list forks of a repository.

I have an organization with a private repository that has been forked by a member of this organization that has a GitHub App with read repository metadata permissions installed. 

I'm querying the `/forks` endpoint with 2 different tokens: 

1. Personal access token of a member of the organization and is NOT the member that has forked the repository.
2. GitHub App installation token.

In the first case, I receive an array that contains the fork owned by another member, in the second the endpoint returns an empty array.

Is there a permission I am missing from the GitHub App that is not listed in the documentation?

Thanks,
Yury

**Comment by Naimulnobel** [comment]
The behavior you're observing is expected and is due to privacy restrictions. 

When you use a personal access token, you're authenticating as a user and you can see all the forks you have access to, which includes public forks and private forks where you have been granted access.

However, when you're authenticating as a GitHub App, you're not acting as a user. A GitHub App only has access to resources which are either owned by the account which installed the app, or are accessible because of the permissions granted during installation. 

In this case, the GitHub App does not have access to the forked repository because it's owned by a different account (the member who forked it), even though the app has read access to the original repository. 

This is a privacy measure to ensure that installing a GitHub App on one account doesn't inadvertently grant access to resources owned by other accounts.

Unfortunately, there's no permission you can add to the GitHub App to change this behavior. The member who owns the fork would need to install your GitHub App on their account to give it access to the forked repository.

**Comment by yury-imbro-ox** [comment]
> The behavior you're observing is expected and is due to privacy restrictions.
> 
> When you use a personal access token, you're authenticating as a user and you can see all the forks you have access to, which includes public forks and private forks where you have been granted access.
> 
> However, when you're authenticating as a GitHub App, you're not acting as a user. A GitHub App only has access to resources which are either owned by the account which installed the app, or are accessible because of the permissions granted during installation.
> 
> In this case, the GitHub App does not have access to the forked repository because it's owned by a different account (the member who forked it), even though the app has read access to the original repository.
> 
> This is a privacy measure to ensure that installing a GitHub App on one account doesn't inadvertently grant access to resources owned by other accounts.
> 
> Unfortunately, there's no permission you can add to the GitHub App to change this behavior. The member who owns the fork would need to install your GitHub App on their account to give it access to the forked repository.

The PAT I've been using does not have access to the other user's fork, so the behavior is inconsistent in that regard
